api/v1beta1/environment_types.go

```diff
@@ -0,0 +1,217 @@
+	TargetNamespace string `json:"targetNamespace"`
+
+	// APIURL is a reference to a cluster API url defined within the kube config file of the cluster credentials secret.
+	APIURL string `json:"apiURL"`
+
+	// IngressDomain is the cluster's ingress domain.
+	// For example, in minikube it would be $(minikube ip).nip.io and in OCP it would look like apps.xyz.rhcloud.com.
+	// If clusterType == "Kubernetes", ingressDomain is mandatory and is enforced by the webhook validation
+	IngressDomain string `json:"ingressDomain,omitempty"`
+
+	// ClusterCredentialsSecret is a reference to the name of k8s Secret, defined within the same namespace as the Environment resource,
+	// that contains a kubeconfig.
+	// The Secret must be of type 'managed-gitops.redhat.com/managed-environment'
+	//
+	// See this temporary URL for details:
+	// https://github.com/redhat-appstudio/managed-gitops/tree/main/examples/m6-demo#gitopsdeploymentmanagedenvironment-resource
+	ClusterCredentialsSecret string `json:"clusterCredentialsSecret"`
+
+	// Indicates that ArgoCD/GitOps Service should not check the TLS certificate.
+	AllowInsecureSkipTLSVerify bool `json:"allowInsecureSkipTLSVerify"`
+
+	// Namespaces allows one to indicate which Namespaces the Secret's ServiceAccount has access to.
+	//
+	// Optional, defaults to empty. If empty, it is assumed that the ServiceAccount has access to all Namespaces.
+	//
+	// The ServiceAccount that GitOps Service/Argo CD uses to deploy may not have access to all of the Namespaces on a cluster.
+	// If not specified, it is assumed that the Argo CD ServiceAccount has read/write at cluster-scope.
+	// - If you are familiar with Argo CD: this field is equivalent to the field of the same name in the Argo CD Cluster Secret.
+	Namespaces []string `json:"namespaces,omitempty"`
+
+	// ClusterResources is used in conjuction with the Namespace field.
+	// If the Namespaces field is non-empty, this field will be used to determine whether Argo CD should
+	// attempt to manage cluster-scoped resources.
+	// - If Namespaces field is empty, this field is not used.
+	// - If you are familiar with Argo CD: this field is equivalent to the field of the same name in the Argo CD Cluster Secret.
+	//
+	// Optional, default to false.
+	ClusterResources bool `json:"clusterResources,omitempty"`
+}
+
+// EnvironmentConfiguration contains Environment-specific configurations details, to be used when generating
+// Component/Application GitOps repository resources.
+type EnvironmentConfiguration struct {
+	// Env is an array of standard environment vairables
+	Env []EnvVarPair `json:"env"`
+
+	// Target is used to reference a DeploymentTargetClaim for a target Environment.
+	// The Environment controller uses the referenced DeploymentTargetClaim to access its bounded
+	// DeploymentTarget with cluster credential secret.
+	Target EnvironmentTarget `json:"target,omitempty"`
```

Comment:
The `.spec.configuration.target` field never made much sense where it was. Let's move it to '.spec.target', and rename to 'claim', so that the full name is '.spec.target.claim'.